dist/common/scripts/scylla_coredump_setup

```diff
@@ -54,13 +57,48 @@ ExternalSizeMax=1024G
         with open('/etc/systemd/coredump.conf', 'w') as f:
             conf = f.write(conf_data)
         if args.dump_to_raiddir:
-            rmtree('/var/lib/systemd/coredump')
+            dot_mount = '''
+[Unit]
+Description=Save coredump to scylla data directory
+Conflicts=umount.target
+Before=scylla-server.service
+After=local-fs.target
+
+[Mount]
+What=/var/lib/scylla/coredump
+Where=/var/lib/systemd/coredump
+Type=none
+Options=bind
+
+[Install]
+WantedBy=multi-user.target
+'''[1:-1]
+            with open('/etc/systemd/system/var-lib-systemd-coredump.mount', 'w') as f:
+                f.write(dot_mount)
             makedirs('/var/lib/scylla/coredump')
-            os.symlink('/var/lib/scylla/coredump', '/var/lib/systemd/coredump')
-        run('systemctl daemon-reload')
+            systemd_unit.reload()
+            systemd_unit('var-lib-systemd-coredump.mount').enable()
         if os.path.exists('/usr/lib/sysctl.d/50-coredump.conf'):
             run('sysctl -p /usr/lib/sysctl.d/50-coredump.conf')
         else:
             with open('/etc/sysctl.d/99-scylla-coredump.conf', 'w') as f:
                 f.write('kernel.core_pattern=|/usr/lib/systemd/systemd-coredump %p %u %g %s %t %e"')
             run('sysctl -p /etc/sysctl.d/99-scylla-coredump.conf')
+
+        fp = tempfile.NamedTemporaryFile()
+        fp.write(b'kill -SEGV $$')
+        fp.flush()
+        p = subprocess.Popen(['/bin/bash', fp.name], stdout=subprocess.PIPE)
+        pid = p.pid
+        p.wait()
+        fp.close()
+
+        print('Generating coredump to test systemd-coredump...\n')
+        # need to wait for systemd-coredump to complete collecting coredump
+        time.sleep(3)
```

Comment:
This looks fragile to me. Shouldn't we run `coredumpctl` in a loop but time-out after, say 10 seconds? @avikivity thoughts?

Reply:
@penberg I think if we run ```coredumpctl info <pid>``` in loop and wait for finishing coredump, the script might run infinitely when  systemd-coredump does not working correctly, ex: failing to install kernel.core_pattern.

We need to show error message on such case, but the case coredump unfinished and the case systemd-coredump not working might return same error: No coredumps found.